src/tasks/ts/tokens.ts

```diff
@@ -0,0 +1,130 @@
+import WethNetworks from "canonical-weth/networks.json";
+import {
+  BigNumber,
+  BigNumberish,
+  Contract,
+  ContractTransaction,
+  providers,
+  Signer,
+} from "ethers";
+import { HardhatRuntimeEnvironment } from "hardhat/types";
+
+import { SupportedNetwork, isSupportedNetwork } from "./deployment";
+
+export interface Erc20Token {
```

Comment:
Why the name change? Wasn't this `TokenDetails` before?

Reply:
[Here](https://github.com/gnosis/gp-v2-contracts/pull/734#discussion_r667745423) is the context of the name change. `TokenDetails` has become confusing since now a "token" could be either an erc20 token or the native token. If this were Rust, I'd definitely have created an `enum Token` with the two  variants `Erc20` and `Native`.